src/main/java/org/dataloader/DataLoaderHelper.java

```diff
@@ -296,15 +298,12 @@ private void possiblyClearCacheEntriesOnExceptions(List<K> keys) {
             return futureCache.get(cacheKey);
         }
 
-        CompletableFuture<V> loadCallFuture;
-        synchronized (dataLoader) {
```

Comment:
This is a weird PR because you gave no explanation of why this synchronised call is not needed.

Are you sayin that a `synchronized` is already in place when this is called??

the loader queue is the key bit of mutable state in the DataLoader (the list of promises to get data at some future batch time) and hence needs synchronization of some sort.

Reply:
My point is - explanation is important

Reply:
@bbakerman Thanks for your response.

Yes. As `@GuardedBy("dataLoader") ` describe, the thread is already hold `dataLoader ` lock when `loadFromCache`  is called.
https://github.com/graphql-java/java-dataloader/blob/4253050b302c6e342a42a67bdab094844ba643ff/src/main/java/org/dataloader/DataLoaderHelper.java#L102